scipy/linalg/src/_linalg_solve.hh

```diff
@@ -276,6 +318,11 @@ _solve(PyArrayObject* ap_Am, PyArrayObject *ap_b, T* ret_data, St structure, int
     npy_intp *strides_b = PyArray_STRIDES(ap_b);
     npy_intp nrhs = PyArray_DIM(ap_b, ndim_b -1); // Number of right-hand-sides
 
+    // Only used for `banded`
+    npy_intp *kls = NULL;
+    npy_intp *kus = NULL;
+    T *ab = NULL;
```

Comment:
ISTM it will be cleaner to have a separate `_solve`-like function just for the banded mode. Then you just allocate exactly what needs allocating, you know exactly the structure etc. The code duplication will be relatively mild; the code will be simpler and easier to reason about.